src/cmds/qdel.c

```diff
@@ -241,25 +341,23 @@ char **envp;
 			pbs_strncpy(warg, warg1, sizeof(warg));
 		}
 
-		stat = pbs_deljob(connect, job_id_out, warg);
-
-		/*
-		 * The counter num_deleted should not be updated  when a history job is deleted .
-		 */
-		if (pbs_errno != PBSE_HISTJOBDELETED)
-			num_deleted++;
-		if (stat && (pbs_errno != PBSE_UNKJOBID && pbs_errno != PBSE_HISTJOBDELETED)) {
-			prt_job_err("qdel", connect, job_id_out);
-			any_failed = pbs_errno;
-		} else if (stat && (pbs_errno == PBSE_UNKJOBID) && !located) {
-			located = TRUE;
-			if (locate_job(job_id_out, server_out, rmt_server)) {
-				pbs_disconnect(connect);
-				pbs_strncpy(server_out, rmt_server, sizeof(server_out));
-				goto cnt;
+		p_delstatus = pbs_deljoblist(connect, svr_itr->jobids, warg);
+		
+		while (p_delstatus != NULL) {
+			if (p_delstatus->code == PBSE_UNKJOBID) {
+				if (locate_job(p_delstatus->name, server_out, rmt_server))
+					add_jid_to_list(rmt_server, rmt_server, &svr_itr);
+				continue;
```

Comment:
calling continue without updating the variable will trigger an infinite loop.

Reply:
Changed to for loop, it's okay to call continue now.